conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/Enclave.kt

```diff
@@ -242,6 +245,18 @@ abstract class Enclave {
     @PotentialPackagePrivate
     private fun initialise(env: EnclaveEnvironment) {
         this.env = env
+
+        // Prevent users from printing to the console when running the enclave in release mode
+        // GraalVM already discards the output so the following if statement is only relevant for GramineSGX
+        // N.B. The developer of the enclave can still override the configuration below and print to the console
+        // when running the enclave in release mode. But that has to be a conscious decision.
```

Comment:
This last statement is not correct. Can you show me how they could do that

Reply:
`package com.r3.conclave.sample.enclave

import java.io.FileDescriptor
import java.io.FileOutputStream
import java.io.PrintStream

/**
 * Simply reverses the bytes that are passed in.
 */
class ReverseEnclave : com.r3.conclave.enclave.Enclave() {
    // We store the previous result to showcase that the enclave internals can be examined in a mock test.
    lateinit var previousResult: ByteArray
    protected override fun receiveFromUntrustedHost(bytes: ByteArray): ByteArray? {
        // This is used for host->enclave calls so we don't have to think about authentication.
        val input = String(bytes)
        val result: ByteArray = reverse(input).toByteArray()
        previousResult = result
        println("Hello. Stdout")
        java.lang.System.err.println("Hello. Stderr")
        return result
    }

    @kotlin.Throws(java.lang.Exception::class)
    protected override fun receiveMail(mail: com.r3.conclave.mail.EnclaveMail, routingHint: String?) {
        // This is used when the host delivers a message from the client.
        // First, decode mail body as a String.
        val stringToReverse = String(mail.getBodyAsBytes())
        // Reverse it and re-encode to UTF-8 to send back.
        val reversedEncodedString: ByteArray = reverse(stringToReverse).toByteArray()
        // Get the post office object for responding back to this mail and use it to encrypt our response.
        val responseBytes: ByteArray = postOffice(mail).encryptMail(reversedEncodedString)
        postMail(responseBytes, routingHint)
    }

    companion object {
        @kotlin.Throws(java.lang.Exception::class)
        private fun reverse(input: String): String {
            System.setOut(PrintStream(FileOutputStream(FileDescriptor.out)))
            println("Hello. Stdout")
            java.lang.System.err.println("Hello. Stderr")
            if(true) {
                throw  Exception("askjdhfksd")
            }
            val builder = StringBuilder(input.length)
            for (i in input.length - 1 downTo 0) builder.append(input[i])
            return builder.toString()
        }
    }
}`

Reply:
The reason I say that is bc the user doesn't have a reference to the stdout/stderr objects to call the set methods with to resign stdout/stderr.

Reply:
I see. I didn't know about `FileDescriptor.out`. Fair enough.

Reply:
In that case, can we also turn off the enclave process redirect to the host. It's not fully secure, but it at least avoids issues if the user is using Conclave as-is and tries to do this trick.